File: Dot examples/ASP Book Examples/Book Examples/Chapter34/cs/App_Code/Entities/BlogComment.cs
using System;
using System.Collections;
using System.Web;
using System.Linq;
using System.Data.Linq.Mapping;
using System.Collections.Generic;


/// <summary>
/// Summary description for BlogComment
/// </summary>
public partial class BlogComment: EntityBase<BlogComment>
{
    public static IEnumerable Select(int blogId, int startRowIndex, int maximumRows)
    {
        return Table.Where(c => c.BlogId == blogId).OrderBy( c => c.Id)
        .Join
            (
                WebUser.Table,
                c => c.AuthorUserName,
                w => w.UserName,
                (c, w) => new {Id = c.Id, Title = c.Title, Comment = c.Comment, AuthorUserName = c.AuthorUserName, Version = c.Version, DateCreated = c.DateCreated, FirstName = w.FirstName, LastName = w.LastName, WebsiteUrl = w.WebsiteUrl}
            )
        .OrderByDescending(a => a.Id)     
        .Skip(startRowIndex)
        .Take(maximumRows);  
    }
    
    
    public static int SelectCountCached(int blogId)
    {
        return Table.Where(c => c.BlogId == blogId).Count();
    }




    protected override void Validate()
    {
        // Security check
        // Only the administrator or original author can modify this comment
        if (!ValidationUtility.IsInRole("Administrators"))
        {
            if (!ValidationUtility.IsUserName(AuthorUserName))
                ValidationErrors.Add("AuthorUserName", "Only the original author or administrator can modify this comment"); 
        }

        // Check for required
        if (!ValidationUtility.SatisfiesRequired(Title))
            ValidationErrors.Add("Title", "Title is required", "Required");

        if (!ValidationUtility.SatisfiesRequired(Comment))
            ValidationErrors.Add("Comment", "Comment is required", "Required");

    }
}

File: Dot examples/ASP Book Examples/Book Examples/Chapter34/vb/App_Code/Entities/BlogComment.vb
Imports System
Imports System.Collections
Imports System.Web
Imports System.Linq
Imports System.Data.Linq.Mapping
Imports System.Collections.Generic


''' <summary>
''' Summary description for BlogComment
''' </summary>
Partial Public Class BlogComment
     Inherits EntityBase(Of BlogComment)

    Public Shared Function [Select](ByVal blogId As Integer, ByVal startRowIndex As Integer, ByVal maximumRows As Integer) As IEnumerable
        Return Table.Where(Function(c) c.BlogId = blogId).OrderBy(Function(c) c.Id).Join(WebUser.Table, Function(c) c.AuthorUserName, Function(w) w.UserName, Function(c, w) New With {.Id = c.Id, .Title = c.Title, .Comment = c.Comment, .AuthorUserName = c.AuthorUserName, .Version = c.Version, .DateCreated = c.DateCreated, .FirstName = w.FirstName, .LastName = w.LastName, .WebsiteUrl = w.WebsiteUrl}).OrderByDescending(Function(a) a.Id).Skip(startRowIndex).Take(maximumRows)
    End Function


    Public Shared Function SelectCountCached(ByVal blogId As Integer) As Integer
        Return Table.Where(Function(c) c.BlogId = blogId).Count()
    End Function




    Protected Overrides Sub Validate()
        ' Security check
        ' Only the administrator or original author can modify this comment
        If Not ValidationUtility.IsInRole("Administrators") Then
            If Not ValidationUtility.IsUserName(AuthorUserName) Then
                ValidationErrors.Add("AuthorUserName", "Only the original author or administrator can modify this comment")
            End If
        End If

        ' Check for required
        If Not ValidationUtility.SatisfiesRequired(Title) Then
            ValidationErrors.Add("Title", "Title is required", "Required")
        End If

        If Not ValidationUtility.SatisfiesRequired(Comment) Then
            ValidationErrors.Add("Comment", "Comment is required", "Required")
        End If
    End Sub
End Class

